tests/extra/test_job_management.py

```diff
@@ -0,0 +1,161 @@
+        requests_mock.get("http://foo.test/", json={"api_version": "1.1.0"})
+        requests_mock.get("http://bar.test/", json={"api_version": "1.1.0"})
+
+        def mock_job_status(job_id, queued=1, running=2):
+            """Mock job status polling sequence"""
+            response_list = sum(
+                [
+                    [
+                        {
+                            "json": {
+                                "id": job_id,
+                                "title": f"Job {job_id}",
+                                "status": "queued",
+                            }
+                        }
+                    ]
+                    * queued,
+                    [
+                        {
+                            "json": {
+                                "id": job_id,
+                                "title": f"Job {job_id}",
+                                "status": "running",
+                            }
+                        }
+                    ]
+                    * running,
+                    [
+                        {
+                            "json": {
+                                "id": job_id,
+                                "title": f"Job {job_id}",
+                                "status": "finished",
+                            }
+                        }
+                    ],
+                ],
+                [],
+            )
+            for backend in ["http://foo.test", "http://bar.test"]:
+                requests_mock.get(f"{backend}/jobs/{job_id}", response_list)
+                # It also needs the job results endpoint, though that can be a dummy implementation.
+                # When the job is finished the system tries to download the results and that is what
+                # needs this endpoint.
+                requests_mock.get(
+                    f"{backend}/jobs/{job_id}/results", json={"links": []}
+                )
+
+        mock_job_status("job-2018", queued=1, running=2)
+        mock_job_status("job-2019", queued=2, running=3)
+        mock_job_status("job-2020", queued=3, running=4)
+        mock_job_status("job-2021", queued=3, running=5)
+        mock_job_status("job-2022", queued=5, running=6)
+
+        manager = MultiBackendJobManager(poll_sleep=0.2)
+        manager.add_backend("foo", connection=openeo.connect("http://foo.test"))
+        manager.add_backend("bar", connection=openeo.connect("http://bar.test"))
+
+        df = pd.DataFrame(
+            {
+                "year": [2018, 2019, 2020, 2021, 2022],
+                # Use simple points in WKT format to test conversion to the geometry dtype
+                "geometry": ["POINT (1 2)"] * 5,
+            }
+        )
+        output_file = tmp_path / "jobs.csv"
+
+        def start_job(row, connection, **kwargs):
+            year = row["year"]
+            return RESTJob(job_id=f"job-{year}", connection=connection)
```

Comment:
FTY: `RESTJob` is deprecated, use `BatchJob` instead

Reply:
Done.